srcalc/src/main/webapp/WEB-INF/views/vistaLoginForm.jsp

```diff
@@ -5,10 +5,21 @@
 
 <section>
     <h2>Please Login</h2>
+    <c:choose>
     <%-- This parameter name must match the Spring Security configuration. --%>
-    <c:if test="${!empty param.login_failure}">
+    <c:when test="${empty param.login_failure}">
+    <%-- Only attempt CCOW auth if this is our first attempt. --%>
+    <p id="attemptingCcowMessage">Attempting Single Sign-On. Please wait...</p>
+    <!-- Using some HTML5-obsolete technology here, but it's the only way to CCOW for now. -->
+    <object id="ContextorControl"
+        data="data:application/x-oleobject;base64,96x4h6lc0xGHJwBgsLXhNwADAADYEwAA2BMAAA=="
+        classid="CLSID:8778ACF7-5CA9-11D3-8727-0060B0B5E137" name="ContextorControl">
```

Comment:
I have no idea what the data and classid values are for. I think they deserve a comment.